File: backend/src/main/java/com/williamdsw/semsys/domain/dto/PersonDTO.java
package com.williamdsw.semsys.domain.dto;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import org.hibernate.validator.constraints.Length;
import com.williamdsw.semsys.domain.Person;

public class PersonDTO implements Serializable {
	// FIELDS

	private static final long serialVersionUID = 1L;
	protected Integer id;

	@NotEmpty(message = "Name is required")
	@Length(min = 5, max = 120, message = "Name must be between 5 and 120 characters")
	protected String name;

	@NotEmpty(message = "Email is required")
	@Email(message = "Invalid email")
	protected String email;

	protected String type;
	protected Set<String> profiles = new HashSet<>();

	// CONSTRUCTORS

	public PersonDTO() {
	}

	public PersonDTO(Person person) {
		super();
		this.id = person.getId();
		this.name = person.getName();
		this.email = person.getEmail();
		this.type = person.getClass().getSimpleName();
		person.getProfiles().forEach(profile -> {
			this.profiles.add(profile.toString());
		});
	}

	// GETTERS / SETTERS

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public Set<String> getProfiles() {
		return profiles;
	}

	public void setProfiles(Set<String> profiles) {
		this.profiles = profiles;
	}
}

File: frontend/src/app/models/domain/person.dto.ts

export class PersonDTO {

    // FIELDS

    private id: number;
    private name: string;
    private email: string;
    private type: string;
    private profiles: string[] = [];
    private imageUrl?: string;
    
    // CONSTRUCTOR

    constructor() {}

    // GETTERS SETTERS

    public getId(): number {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getName(): string {
        return this.name;
    }

    public setName(name: string) {
        this.name = name;
    }

    public getEmail(): string {
        return this.email;
    }

    public setEmail(email: string) {
        this.email = email;
    }

    public getType(): string {
        return this.type;
    }

    public setType(type: string) {
        this.type = type;
    }

    public getProfiles(): string[] {
        return this.profiles;
    }

    public setProfiles(profiles: string[]) {
        this.profiles = profiles;
    }

    public getImageUrl(): string {
        return this.imageUrl;
    }

    public setImageUrl(imageUrl: string) {
        this.imageUrl = imageUrl;
    }
}
